healthcare.management.system/rest-module/src/main/java/com/healthcare/management/system/rest/controller/UserController.java

```diff
@@ -0,0 +1,64 @@
+package com.healthcare.management.system.rest.controller;
+
+import com.healthcare.management.system.rest.entity.User;
+import com.healthcare.management.system.rest.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/api")
+// TODO: 18.08.22 add version in the api endpoint path
+
+public class UserController {
+    private UserService userService;
+
+    @Autowired
+    public  UserController(UserService theUserService) {
+        userService = theUserService;
+    }
+
+    @GetMapping("/users")
+    public List<User> findAllUsers(){
+        return userService.findAll();
+    }
+
+    @GetMapping("/users/{userId}")
+    public User getUser(@PathVariable int userId){
+        User theUser = userService.findById(userId);
+
+        if (theUser == null) {
+            throw new RuntimeException("There is no user with id: " + userId);
+        }
+        return theUser;
+    }
+
+    @PostMapping("/users")
+    public User createUser(@RequestBody User theUser){
+        // just in case they pass an id in JSON ... set id to 0
+        // this is to force a save of new item ... instead of update
+        theUser.setId(0);
+
+        userService.createOrUpdate(theUser);
+        return theUser;
+    }
+
+    @PutMapping("/users")
+    public User updateUser(@RequestBody User theUser){
```

Comment:
Same thing applies for this endpoint as the create user endpoint - use `UserDTO`. 

Also REST rules specify the id should be part of the `PUT` url - /users/{userId}. The rest of the fields should be the same as the create user DTO, so we can re-use it.